building/trusted_applications.rst

```diff
@@ -373,35 +373,50 @@ third step the binary and its signature are stitched together into the full TA.
 Offline Signing of TAs
 ======================
 
-The TA dev kit does sign an application as last step of the linking process. For
-example, the file ``ta/arch/arm/link.mk`` in the :ref:`optee_os` source tree
-contains the statement
+There are two types of TAs that can be signed offline: the in-tree TAs, which come
+with the OP-TEE OS (for example the ``pkcs11`` TA) and any external TA coming from the user. Offline signing
+runs directly on `.stripped.elf` TA files and there is a slight difference as how they are
+generated. The in-tree `.stripped.elf` TA files are generated during the OP-TEE OS compilation, while
+with external TAs it is usually done by the user. In both cases however, the signing process is the
+same.
+
+Offline signing is done with the following sequence of steps:
+
+0. (Preparation) Generate a 2048 bit RSA key for signing in a secure, offline
+environment and extract the public key. For example:
 
 .. code-block:: sh
 
-    $(q)$(SIGN) --key $(TA_SIGN_KEY) --uuid $(user-ta-uuid) \
-	    --in $$< --out $$@
+    openssl genrsa -out rsa2048.pem 2048
+    openssl rsa -in rsa2048.pem -pubout -out rsa2048_pub.pem 
 
-To avoid build errors when signing offline, this make script needs to be
-adopted. The signing script can be found at
-``$(TA_DEV_KIT_DIR)/../scripts/sign_encrypt.py``
 
-Overall, offline signing is done with the following sequence of steps:
+1. Build the OP-TEE OS with the variable ``TA_PUBLIC_KEY`` set to the public
+key generated above
 
-0. (Preparation) Generate a 2048 bit RSA key for signing in a secure, offline
-environment. Extract the public key and copy it to the ``keys`` directory in the
-:ref:`optee_os` source tree. Adjust ``TA_SIGN_KEY`` for different file/path
-names. (Copy and) modify the ``link.mk`` file for the default linking step to
-produce a digest of the TA binary instead of the full TA.
+.. code-block:: sh
+
+    TA_PUBLIC_KEY=/path/to/public_key.pem make all
+
+The building will do two things:
+
+* | It will embed the ``TA_PUBLIC_KEY`` key into the OP-TEE core image, which will be used to
+  | authenticate the TAs.
+
+* | It will generate `.stripped.elf` files of the in-tree TAs and sign them with the dummy key
+  | pointed by ``TA_SIGN_KEY``, thus creating `.ta` files. Note that the generated `.ta` files are not
+  | to be used as they are not compatible with the public key embedded into the OP-TEE core image. 
+  
+Note that the rest of the steps below use `.stripped.elf` files, either generated by building the
```

Comment:
files, either ... user. -> files.

Reply:
Take a look at [this](https://github.com/OP-TEE/optee_docs/pull/125/commits/da5d12ef0465fa425a2b41638979a8b48e81fbfb) commit.

Reply:
Ok.